app/code/Magento/InventoryConfigurableProduct/Ui/DataProvider/Product/Form/Modifier/InventoryConfigurablePanel.php

```diff
@@ -16,8 +18,27 @@
  */
 class InventoryConfigurablePanel extends AbstractModifier
```

Comment:
Looks like we need to move this to modifier `<type name="Magento\ConfigurableProduct\Ui\DataProvider\Product\Form\Modifier\Composite">`
And remove `AllowedProductTypes` check

Reply:
@naydav done